generated/nidigital/session.py

```diff
@@ -2871,14 +2871,31 @@ def write_source_waveform_site_unique(self, waveform_name, waveform_data):
         for site in waveform_data:
             if len(waveform_data[site]) != actual_samples_per_waveform:
                 raise ValueError('Mismatched length of waveforms. All must be the same length.')
-            if waveform_data[site].typecode != 'L':
-                raise ValueError('Wrong array element type. Must be unsigned 32 bit int ("L"), was {}'.format(waveform_data[site].typecode))
+            # Check the type by using string comparison so that we don't import numpy unecessarilly.
+            if str(type(waveform_data[site])).find("'numpy.ndarray'") != -1:
+                import numpy
+                if waveform_data[site].dtype == numpy.int32:
+                    wfm = array.array('L', waveform_data[site])
```

Comment:
Theoretical maximum is 32 MB, but users typically load a large number of small waveforms.
Maybe we can use `numpy.ctypeslib.as_ctypes`?

Reply:
We use a memoryview to put together the contiguous memory we need to pass into the DLL. Each site waveform needs to be array.array('L') to work.

This means if you want to avoid an extra copy, use array.array('L').